### PR TITLE
Fixes an issue with xeno cooldown abilities

### PR DESCRIPTION
**Pull request by Rohesie**
Also removes hopefully the last runtime from the Shrike.

In short, `add_cooldown()` was returning if there were any `active_timers`. Which meant that an ability with a `CALLBACK` would simply fail to add cooldown.
Since `on_cooldown` will always be `TRUE` when the timer is running and `FALSE` otherwise, this will avoid the runtimes the original check wanted to.

Fixed a few other places where it would take or delete any timer disrespecting ids.

### CODE REVIEW

File: code/datums/actions/xeno_action.dm [hunk 1]
```diff
@@ -114,27 +114,31 @@
...
 /datum/action/xeno_action/proc/cooldown_remaining()
-	for(var/i in active_timers)
-		var/datum/timedevent/timer = i
-		return (timer.timeToRun - world.time) * 0.1
-	return 0
+	return timeleft(cooldown_id) * 0.1
+
 
 //override this for cooldown completion.
 /datum/action/xeno_action/proc/on_cooldown_finish()
```
Comment (spookydonut) [inline code/datums/actions/xeno_action.dm]: clear the cooldown_id here

Comment (Rohesie) [reply code/datums/actions/xeno_action.dm]: Removed `on_cooldown` in favor of `cooldown_id`

File: code/datums/actions/xeno_action.dm [hunk 2]
```diff
@@ -112,33 +111,36 @@
...
-	on_cooldown = FALSE
+	cooldown_id = null
```
Comment (DominikPanic) [inline code/datums/actions/xeno_action.dm]:
```suggestion
	qdel(cooldown_id)
```
to actually stop the timer properly

Comment (Rohesie) [reply code/datums/actions/xeno_action.dm]: This proc is called by `clear_cooldown()`, which stops the timer.

Comment (Rohesie) [reply code/datums/actions/xeno_action.dm]: May be safer to clear it there, though...

Comment (DominikPanic) [reply code/datums/actions/xeno_action.dm]: then it should be done like that there, it's bad to split these two lines apart

Comment (spookydonut) [reply code/datums/actions/xeno_action.dm]: this suggestion is wrong since on_cooldown_finish is called when the timer ends

Comment (spookydonut) [reply code/datums/actions/xeno_action.dm]: This is coded correctly no need to change anything now.

**Pull request merged**